### PR TITLE
Improve build performance

### PR DESCRIPTION
**Pull request by piotrpMSFT**
Fixes issue #698 

Cross-gen Stage1 to reduce Stage 2 compilation from 13 mins to 3 minutes
Eliminate differences between compiling stage 1 and stage 2 to reduce complexity

### CODE REVIEW

File: scripts/compile/compile-stage.ps1 [hunk 1]
```diff
@@ -84,3 +84,18 @@ $BinariesForCoreHost | ForEach-Object {
     mv $OutputDir\bin\$_.exe $OutputDir\bin\$_.dll
     cp $OutputDir\bin\corehost.exe $OutputDir\bin\$_.exe
 }
+
+# Crossgen Roslyn
+header "Crossgening Roslyn compiler ..."
+_cmd "$RepoRoot\scripts\crossgen\crossgen_roslyn.cmd ""$OutputDir"""
+
+# Copy dnx into stage 2
```
Comment (livarcocc) [inline scripts/compile/compile-stage.ps1]: Is this comment still correct? it says stage 2 here.

**Comment by livarcocc** [comment]
LGTM

**Comment by brthor** [comment]
LGTM

File: scripts/compile/compile-stage.ps1 (hunk 1 above)
Comment (piotrpMSFT) [inline scripts/compile/compile-stage.ps1]: fixed

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu Debug Build please

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu Release Build please

**Pull request merged**